test/spmd/tensor/parallel/test_parallelize_api.py

```diff
@@ -79,59 +79,64 @@ def test_creat_1d_device_mesh_error(self):
-            for name, param in local_module.named_parameters():
-                dist_param = dist_module.get_parameter(name)
+        for name, param in local_module.named_parameters():
+            dist_param = dist_module.get_parameter(name)
+            param = param.grad if compare_grad else param
+            dist_param = dist_param.grad if compare_grad else dist_param
+            if (
+                self.rank == 0
+                or not skip_rowwise_bias
+                or name not in ["bias", "net2.bias"]
+            ):
                 self.assertEqual(
                     param,
                     dist_param.redistribute(
-                        device_mesh=device_mesh, placements=replicate
+                        device_mesh=dist_param.device_mesh, placements=replicate
                     ).to_local(),
                 )
 
-            # clear gradients
-            local_optim.zero_grad()
-            dist_optim.zero_grad()
-
-            # check forward correctness
-            local_output = local_module(local_input)
-            dist_output = dist_module(dist_input)
-            self.assertEqual(local_output, dist_output.to_local())
+    def _compare_module(
+        self, local_module, dist_module, inp_size, rowwise=False
+    ):
+        LR = 0.25  # the learning rate we use for testing
+        local_optim = torch.optim.SGD(local_module.parameters(), lr=LR)
+        dist_optim = torch.optim.SGD(dist_module.parameters(), lr=LR)
+        torch.manual_seed(0)
+        inp = torch.rand(*inp_size, device=self.device_type)
+        self._compare_params(local_module, dist_module)
+
+        # check forward correctness
+        local_output = local_module(inp)
+        inp = inp.chunk(self.world_size, dim=-1)[self.rank] if rowwise else inp
+        dist_output = dist_module(inp)
+        dist_output = (
+            dist_output.to_local()
+            if isinstance(dist_output, DTensor)
+            else dist_output
+        )
+        self.assertEqual(local_output, dist_output)
 
-            local_output.sum().backward()
-            dist_output.sum().backward()
+        local_output.sum().backward()
+        dist_output.sum().backward()
 
-            # check backward correctness:
-            #   ensure gradients are same
-            for name, param in local_module.named_parameters():
-                dist_param = dist_module.get_parameter(name)
-                self.assertEqual(
-                    param.grad,
-                    dist_param.grad.redistribute(
-                        device_mesh=device_mesh, placements=replicate
-                    ).to_local(),
-                )
+        # check backward and ensure gradients are same
+        self._compare_params(local_module, dist_module, rowwise, True)
 
-            local_optim.step()
-            dist_optim.step()
+        local_optim.step()
+        dist_optim.step()
+        self._compare_params(local_module, dist_module, rowwise, True)
```

Comment:
True -> False since we want to check correctness of param update?

Reply:
opps.